### PR TITLE
Added the FARGATE support

### PR DESCRIPTION
**Pull request by shashimal**
- Exposed the launch_type argument, so the users can provide FARGATE type. 
- Made some of the mandatory arguments to optional as they were only required by EC2 type.

### CODE REVIEW

**Review by niroz89 — COMMENTED** [review]
@shashimal can we change if the fargate launch type, let the module to not create the launch template.

File: modules/autoscaling-group/main.tf [hunk 1]
```diff
@@ -19,7 +19,7 @@ module "asg" {
   vpc_zone_identifier             = var.subnets
 
   # Launch template
-  create_launch_template      = true
+  create_launch_template      = var.create_launch_template && var.launch_type == "EC2"
```
Comment (gowripadmajas) [inline modules/autoscaling-group/main.tf:22]: @shashimal ,if Launch type is EC2, there should definitely be a Launch_template. I feel condition on launch_type alone is enough.

Comment (gowripadmajas) [inline modules/autoscaling-group/main.tf:22]: @shashimal ,if Launch type is EC2, there should definitely be a Launch_template. I feel condition on launch_type alone is enough.

**Review by gowripadmajas — COMMENTED** [review]
@shashimal ,if Launch type is EC2, there should definitely be a Launch_template. I feel condition on launch_type alone is enough.

File: modules/autoscaling-group/main.tf (hunk 1 above)
Comment (niroz89) [reply modules/autoscaling-group/main.tf:22]: @gowripadmajas , guess this can be used if we are providing existing launch template.

**Review by niroz89 — APPROVED** [review]

**Review by wayne-root — APPROVED** [review]

**Pull request merged**